src/test/scala/edu/ie3/simona/agent/grid/DBFSAlgorithmParticipantSpec.scala

```diff
@@ -0,0 +1,399 @@
+                     |akka.loglevel="OFF"
+        """.stripMargin)
+      )
+    )
+    with DBFSMockGridAgents
+    with ConfigTestData
+    with ImplicitSender
+    with DbfsTestGridWithParticipants {
+
+  private val scheduler = TestProbe("scheduler")
+  private val primaryService = TestProbe("primaryService")
+  private val weatherService = TestProbe("weatherService")
+
+  private val environmentRefs = EnvironmentRefs(
+    scheduler = scheduler.ref,
+    primaryServiceProxy = primaryService.ref,
+    weather = weatherService.ref,
+    evDataService = None
+  )
+
+  protected val resultListener: TestProbe = TestProbe("resultListener")
+
+  private val superiorGridAgent = SuperiorGA(
+    TestProbe("superiorGridAgent_1000"),
+    Seq(supNodeA.getUuid)
+  )
+
+  "Test participant" should {
+
+    val gridAgentWithParticipants = system.actorOf(
+      GridAgent.props(
+        environmentRefs,
+        simonaConfig,
+        Iterable(resultListener.ref)
+      )
+    )
+
+    s"initialize itself when it receives a $InitializeGridAgentTrigger with corresponding data" in {
+      val triggerId = 0
+      val loadAgentTriggerId = 1
+
+      // this subnet has 1 superior grid (HöS) and 3 inferior grids (MS). Map the gates to test probes accordingly
+      val subGridGateToActorRef: Map[SubGridGate, ActorRef] =
+        hvSubGridGates.map {
+          case gate if gate.getInferiorSubGrid == hvGridContainer.getSubnet =>
+            gate -> superiorGridAgent.ref
+        }.toMap
+
+      val gridAgentInitData =
+        GridAgentInitData(
+          hvGridContainer,
+          subGridGateToActorRef,
+          RefSystem("2000 MVA", "110 kV")
+        )
+
+      // send init data to agent and expect a CompletionMessage
+      scheduler.send(
+        gridAgentWithParticipants,
+        TriggerWithIdMessage(
+          InitializeGridAgentTrigger(gridAgentInitData),
+          triggerId,
+          gridAgentWithParticipants
+        )
+      )
+
+      val (loadAgent, initializeTrigger): (
+          ActorRef,
+          InitializeParticipantAgentTrigger[PrimaryData, InitializeStateData[
+            PrimaryData
+          ]]
+      ) =
+        scheduler.expectMsgPF() {
+          case ScheduleTriggerMessage(
+                initializeTrigger: InitializeParticipantAgentTrigger[
+                  PrimaryData,
+                  InitializeStateData[PrimaryData]
+                ],
+                loadAgent
+              ) =>
+            (loadAgent, initializeTrigger)
+        }
+
+      scheduler.expectMsgPF() {
+        case CompletionMessage(
+              triggerId,
+              Some(
+                Vector(
+                  ScheduleTriggerMessage(triggerToBeScheduled, gridAgentActor)
+                )
+              )
+            ) =>
+          triggerId shouldBe 0
+          triggerToBeScheduled shouldBe ActivityStartTrigger(3600L)
+          gridAgentActor shouldBe gridAgentWithParticipants
```

Comment:
You're saving `0L` in the variable `triggerId` (line 83), so why don't you check for that directly? Same for the trigger and actor. So this could be an `expectMsg` instead with message:
```
CompletionMessage(triggerId, Some(Seq(ScheduleTriggerMessage(ActivityStartTrigger(3600L), gridAgentWithParticipants))))
```